src/layers/io/input_layer.cpp

```diff
@@ -77,21 +77,23 @@ void input_layer<T, L, D>::write_specific_proto(lbann_data::Layer& proto) const
 }
 
 template <typename TensorDataType, data_layout T_layout, El::Device Dev>
-void input_layer<TensorDataType, T_layout, Dev>::setup_dims(
-  DataReaderMetaData& dr_metadata)
+void input_layer<TensorDataType, T_layout, Dev>::setup_dims()
 {
-  data_type_layer<TensorDataType>::setup_dims(dr_metadata);
+  data_coordinator& dc = get_trainer().get_data_coordinator();
+  const DataReaderMetaData& dr_metadata = dc.get_dr_metadata();
```

Comment:
A new input layer should avoid this altogether

Reply:
@tbennun I need to go back and revisit why we plumbed this through in the first place.

Reply:
@tbennun to answer my own question, this was a small step to get rid of data readers everywhere.